src/ExpoClient.ts

```diff
@@ -74,13 +75,30 @@ export class Expo {
   async sendPushNotificationsAsync(messages: ExpoPushMessage[]): Promise<ExpoPushTicket[]> {
     const actualMessagesCount = Expo._getActualMessageCount(messages);
 
-    const data = await this.requestAsync(`${BASE_API_URL}/push/send`, {
-      httpMethod: 'post',
-      body: messages,
-      shouldCompress(body) {
-        return body.length > 1024;
+    const data = await promiseRetry(
```

Comment:
Suggestion: move this retry into requestAsync around the fetch call. 

Assume: max concurrency (promise limit) of 6, developer makes 10 concurrent calls to this function, all server responses are 429s.

Currently with this PR: All 10 sendPushNotificationsAsync calls will invoke requestAsync. The first 6 fetch requests will go out in parallel and we'll get 429s back for all of them. The sendPushNotificationsAsync promises for those 6 requests will sleep due to backoff and the remaining 4 sendPushNotificationsAsync calls will immediately make requests while the other 6 are sleeping. That's not what we want.

With proposed suggestion: All 10 sendPushNotificationsAsync calls will invoke requestAsync. The first 6 fetch requests will go out in parallel and we'll get 429s back for all of them and they will backoff, still holding the promise-limit semaphore (so to speak). The other 4 sendPushNotificationsAsync will not be able to send fetch requests until at least 4 of the first 6 have finished all retries.

Relatedly, using mock timers in a unit test to test this behavior would be useful.